### PR TITLE
Add cod4 map to dlc list (and absolutely nothing else)

### PR DESCRIPTION
**Pull request by Rackover**
(The nothing else has been thoroughly tested)

### CODE REVIEW

**Comment by Rackover** [comment]
https://user-images.githubusercontent.com/33836535/209402281-7b76fc7f-0101-4c6c-9d27-cdcacd1ddb11.mp4

File: src/Components/Modules/PlayerMovement.cpp [hunk 1]
```diff
@@ -242,6 +243,20 @@ namespace Components
 
 		BGPlayerCollision = Dvar::Register<bool>("bg_playerCollision",
 			true, Game::DVAR_CODINFO, "Push intersecting players away from each other");
+
+		BGClimbAnything = Dvar::Register<bool>("bg_climbAnything",
+			false, Game::DVAR_CODINFO, "Allows to treat any surface as a ladder");
+	}
+
+	void PlayerMovement::PM_playerTraceForLadder(Game::pmove_s* a1, Game::trace_t* results, const float* start, const float* end, Game::Bounds* bounds, int passEntityNum, int contentMask)
```
Comment (diamante0018) [inline src/Components/Modules/PlayerMovement.cpp:251]: a1 -> pm

File: src/Components/Modules/PlayerMovement.hpp [hunk 2]
```diff
@@ -22,7 +23,8 @@ namespace Components
 		static const Game::dvar_t* BGBouncesAllAngles;
 		static const Game::dvar_t* PlayerDuckedSpeedScale;
 		static const Game::dvar_t* PlayerProneSpeedScale;
-
+		
+		static void PM_playerTraceForLadder(Game::pmove_s* a1, Game::trace_t* results, const float* start, const float* end, Game::Bounds* bounds, int passEntityNum, int contentMask);
```
Comment (diamante0018) [inline src/Components/Modules/PlayerMovement.hpp:27]: Same

**Review by diamante0018 — CHANGES_REQUESTED** [review]

**Comment by Rackover** [comment]
shoutout to @GEEKiDoS  btw for the original idea!!

File: src/Components/Modules/PlayerMovement.cpp [hunk 3]
```diff
@@ -242,6 +243,20 @@ namespace Components
...
+
+		BGClimbAnything = Dvar::Register<bool>("bg_climbAnything",
+			false, Game::DVAR_CODINFO, "Allows to treat any surface as a ladder");
+	}
+
+	void PlayerMovement::PM_playerTraceForLadder(Game::pmove_s* pm, Game::trace_t* results, const float* start, const float* end, Game::Bounds* bounds, int passEntityNum, int contentMask)
+	{
+		Utils::Hook::Call<void(Game::pmove_s*, Game::trace_t*, const float*, const float*, Game::Bounds*, int, int)>(0X458980)(pm, results, start, end, bounds, passEntityNum, contentMask);
+
+#define SURF_LADDER 0x00000008
```
Comment (diamante0018) [inline src/Components/Modules/PlayerMovement.cpp]: constexpr, this is a bad copy-paste from geekidos code.

File: src/Components/Modules/PlayerMovement.cpp [hunk 4]
```diff
@@ -242,6 +243,20 @@ namespace Components
...
 		BGPlayerCollision = Dvar::Register<bool>("bg_playerCollision",
 			true, Game::DVAR_CODINFO, "Push intersecting players away from each other");
+
+		BGClimbAnything = Dvar::Register<bool>("bg_climbAnything",
+			false, Game::DVAR_CODINFO, "Allows to treat any surface as a ladder");
+	}
+
+	void PlayerMovement::PM_playerTraceForLadder(Game::pmove_s* pm, Game::trace_t* results, const float* start, const float* end, Game::Bounds* bounds, int passEntityNum, int contentMask)
+	{
+		Utils::Hook::Call<void(Game::pmove_s*, Game::trace_t*, const float*, const float*, Game::Bounds*, int, int)>(0X458980)(pm, results, start, end, bounds, passEntityNum, contentMask);
```
Comment (diamante0018) [inline src/Components/Modules/PlayerMovement.cpp]: already present in functions.cpp

**Review by diamante0018 — APPROVED** [review]

**Review by diamante0018 — APPROVED** [review]

**Pull request merged**